device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/DeviceIO.java

```diff
@@ -89,7 +89,7 @@
 
 
     // This lock is used to keep calls to open/close/connection status changes synchronous.
-    private Object stateLock = new Object();
+    private final Object stateLock = new Object();
```

Comment:
This change look familiar, haha. This will resolve one of the threading warnings, too